llama_index/indices/managed/google/generativeai/base.py

```diff
@@ -0,0 +1,240 @@
+        **kwargs: Any,
+    ) -> None:
+        """Creates an instance of GoogleIndex.
+
+        Prefer to use the factories `from_corpus` or `create_corpus` instead.
+        """
+        actual_service_context = service_context or google_service_context
+
+        self._store = vector_store
+        self._index = VectorStoreIndex.from_vector_store(
+            vector_store, service_context=actual_service_context, **kwargs
+        )
+
+        super().__init__(
+            index_struct=self._index.index_struct,
+            service_context=actual_service_context,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_corpus(
+        cls: Type[IndexType], *, corpus_id: str, **kwargs: Any
+    ) -> IndexType:
+        """Creates a GoogleIndex from an existing corpus.
+
+        Args:
+            corpus_id: ID of an existing corpus on Google's server.
+
+        Returns:
+            An instance of GoogleIndex pointing to the specified corpus.
+        """
+        _logger.debug(f"\n\nGoogleIndex.from_corpus(corpus_id={corpus_id})")
+        return cls(
+            vector_store=GoogleVectorStore.from_corpus(corpus_id=corpus_id), **kwargs
+        )
+
+    @classmethod
+    def create_corpus(
+        cls: Type[IndexType],
+        *,
+        corpus_id: Optional[str] = None,
+        display_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> IndexType:
+        """Creates a GoogleIndex from a new corpus.
+
+        Args:
+            corpus_id: ID of the new corpus to be created. If not provided,
+                Google server will provide one.
+            display_name: Title of the new corpus. If not provided, Google
+                server will provide one.
+
+        Returns:
+            An instance of GoogleIndex pointing to the specified corpus.
+        """
+        _logger.debug(
+            f"\n\nGoogleIndex.from_new_corpus(new_corpus_id={corpus_id}, new_display_name={display_name})"
+        )
+        return cls(
+            vector_store=GoogleVectorStore.create_corpus(
+                corpus_id=corpus_id, display_name=display_name
+            ),
+            **kwargs,
+        )
+
+    @classmethod
+    def from_documents(
+        cls: Type[IndexType],
+        documents: Sequence[Document],
+        storage_context: Optional[StorageContext] = None,
+        service_context: Optional[ServiceContext] = None,
+        show_progress: bool = False,
+        **kwargs: Any,
+    ) -> IndexType:
+        """Build an index from a sequence of documents."""
+        _logger.debug(f"\n\nGoogleIndex.from_documents(...)")
+
+        new_display_name = f"Corpus created on {datetime.datetime.now()}"
+        c = cls(
+            vector_store=GoogleVectorStore.create_corpus(display_name=new_display_name),
+            **kwargs,
+        )
+
+        index = cast(GoogleIndex, c)
+        index.insert_documents(documents=documents, service_context=service_context)
```

Comment:
does this only support text based documents?
or it can support images as well?

Reply:
For now, just text document.